File: hackerrank/Algorithms/01-Warmup/TheLoveLetterMystery/TheLoveLetterMystery.hs
-- HACKERRANK - The Love-Letter Mystery
-- https://www.hackerrank.com/challenges/the-love-letter-mystery
module Main where

import Control.Monad
import Data.Char

main :: IO ()
main =
    do line <- getLine
       let caseNum = read line :: Int
       subMain caseNum

subMain :: Int -> IO()
subMain caseNum =
    unless  (0 == caseNum) $
    do line <- getLine
       let zipped = take (div (length line) 2) (zip line (reverse line))
           count = sum [abs(ord a - ord b)|(a,b) <- zipped]
       print (id count)
       subMain (caseNum-1)

File: hackerrank/Algorithms/Warmup/TheLoveLetterMystery/TheLoveLetterMystery.cpp
// HACKERRANK - The Love-Letter Mystery
// https://www.hackerrank.com/challenges/the-love-letter-mystery
#include <cstdio>
#include <cstring>
#include <cstdlib>

using namespace std;

int main (int argc, char *argv[]) {
    int T = 0;
    char str[10001] = {0, };

    scanf("%d", &T);

    while (T--) {
        int count=0;
        int st = 0, ed=0;

        scanf("%s", str);
        ed = strlen(str)-1;

        while (st < ed) {
            count += abs(str[st]-str[ed]);
            st++;
            ed--;
        }
        printf("%d\n", count);
    }

    return 0;
}
